# HG changeset patch
# User alta88@fixall.com
# Date 1715706249 -7200
# Parent  e5f70f70fd19a58f9e58a4eb6a242ac4a04449b3
QFB fixes for TB changes in 128:
1. clear button must have a tabindex for keyboard support.
2. clear button must refocus on input element when cleared.
3. focusring broken on qfb buttons with tab stops/buttongroup.

Regressed by or fixes to: Bug 1835687, Bug 1882592, Bug 1866717

diff --git a/mail/base/content/widgets/search-bar.inc.xhtml b/mail/base/content/widgets/search-bar.inc.xhtml
--- a/mail/base/content/widgets/search-bar.inc.xhtml
+++ b/mail/base/content/widgets/search-bar.inc.xhtml
@@ -10,10 +10,9 @@
     <button id="clear-button"
             type="reset"
             class="button button-flat icon-button"
             data-l10n-id="search-bar-clear-button"
-            hidden="hidden"
-            tabindex="-1">
+            hidden="hidden">
       <slot name="clear-button"></slot>
     </button>
     <button id="search-button"
             class="button button-flat icon-button">
diff --git a/mail/base/content/widgets/search-bar.mjs b/mail/base/content/widgets/search-bar.mjs
--- a/mail/base/content/widgets/search-bar.mjs
+++ b/mail/base/content/widgets/search-bar.mjs
@@ -161,11 +161,15 @@ export class SearchBar extends HTMLEleme
   }
 
   /**
    * Reset the search bar to its empty state.
+   * @param {boolean} focus - Whether to focus #input after reset().
    */
-  reset() {
+  reset(focus = true) {
     this.#input.value = "";
+    if (focus) {
+      this.focus();
+    }
     this.#clearButton.hidden = true;
   }
 
   /**
diff --git a/mail/components/storybook/stories/search-bar.stories.mjs b/mail/components/storybook/stories/search-bar.stories.mjs
--- a/mail/components/storybook/stories/search-bar.stories.mjs
+++ b/mail/components/storybook/stories/search-bar.stories.mjs
@@ -34,9 +34,8 @@ const Template = ({ label, disabled }) =
         type="reset"
         data-l10n-id="search-bar-clear-button"
         class="button button-flat icon-button"
         hidden="hidden"
-        tabindex="-1"
       >
         <slot name="clear-button"></slot>
       </button>
       <button id="search-button" class="button button-flat icon-button">
diff --git a/mail/modules/QuickFilterManager.sys.mjs b/mail/modules/QuickFilterManager.sys.mjs
--- a/mail/modules/QuickFilterManager.sys.mjs
+++ b/mail/modules/QuickFilterManager.sys.mjs
@@ -1382,11 +1382,12 @@ export var MessageTextFilter = {
       panel.hidePopup();
     }
 
     // Propagate a cleared text filter to the search bar input.
+    // Don't focus if textfilter is not the active filter.
     const desiredValue = aFilterValue.text || "";
     if (!desiredValue) {
-      aNode.reset();
+      aNode.reset(aNode.id == aMuxer.activeElement?.id);
     }
 
     // Update our expanded filters buttons.
     const states = aFilterValue.states;
diff --git a/mail/themes/shared/mail/search-bar.css b/mail/themes/shared/mail/search-bar.css
--- a/mail/themes/shared/mail/search-bar.css
+++ b/mail/themes/shared/mail/search-bar.css
@@ -49,26 +49,26 @@ input {
   flex-direction: column;
   justify-content: space-around;
 
   &.button-flat.icon-button {
-    padding: 4px;
-    margin: 0;
+    padding: 2px;
 
     &:focus-visible {
-      outline-offset: -1px;
+      outline-style: solid;
+      outline-width: 2px;
+      outline-color: var(--search-focus-outline-color);
+      outline-offset: var(--search-outline-offset);
+      background-color: var(--search-bar-focus-background);
 
       @media (-moz-windows-accent-color-in-titlebar) {
         outline-offset: -4px;
       }
     }
   }
 
   #clear-button& {
-    height: 22px;
     /* Width of the search button. */
     inset-inline-end: 26px;
-    margin-block: auto;
-    padding: 2px;
   }
 }
 
 div {
diff --git a/mail/themes/shared/mail/threadCard.css b/mail/themes/shared/mail/threadCard.css
--- a/mail/themes/shared/mail/threadCard.css
+++ b/mail/themes/shared/mail/threadCard.css
@@ -57,9 +57,9 @@
     --thread-card-border-hover: color-mix(in srgb, var(--selected-item-color) 60%, transparent);
     --thread-card-border-focus: var(--selected-item-color);
     --thread-card-hover: var(--selected-item-color);
     --thread-card-focus: color-mix(in srgb, var(--selected-item-color) 20%, transparent);
-    --focus-outline: transparent;
+    --focus-outline-threadcard: transparent;
     --indicator-background: transparent;
     --indicator-border: transparent;
     --read-status-fill: transparent;
     --read-status-stroke: transparent;
@@ -128,9 +128,9 @@
     --thread-card-border: CanvasText;
     --thread-card-border-hover: SelectedItem;
     --thread-card-hover: transparent;
     --thread-card-focus: SelectedItem;
-    --focus-outline: SelectedItem;
+    --focus-outline-threadcard: SelectedItem;
     --indicator-background: transparent;
     --indicator-border: transparent;
     --read-status-fill: transparent;
     --read-status-stroke: transparent;
@@ -547,9 +547,9 @@
     }
 
     @media (prefers-contrast) {
       &:is(:hover, :focus, :focus-within, :focus-visible, .current) .thread-card-container {
-        outline: 2px solid var(--focus-outline);
+        outline: 2px solid var(--focus-outline-threadcard);
         outline-offset: -3px;
       }
 
       &.current.selected .thread-card-container {
@@ -728,9 +728,9 @@
   }
 
   /* Spam */
   & tr[is="thread-card"][data-properties~="junk"] {
-    --focus-outline: var(--focus-outline-spam);
+    --focus-outline-threadcard: var(--focus-outline-spam);
     --thread-card-border-hover: var(--thread-card-border-hover-spam);
     --thread-card-background-current: var(--thread-card-background-current-spam);
     --thread-card-hover: var(--thread-card-hover-spam);
     --thread-card-background-current: var(--thread-card-background-current-spam);
